pkg/reconciler/buildrun/buildrun.go

```diff
@@ -232,6 +242,23 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 			return reconcile.Result{}, nil
 		}
 
+		if buildRun.IsCanceled() && !lastTaskRun.IsCancelled() {
+			ctxlog.Info(ctx, "buildRun marked for cancellation, patching task run", namespace, request.Namespace, name, request.Name)
+			// patch tekton taskrun a la tkn to start tekton's cancelling logic
+			trueParam := true
+			if err := r.patchTaskRun(ctx, lastTaskRun, "replace", "/spec/status", v1beta1.TaskRunSpecStatusCancelled, metav1.PatchOptions{Force: &trueParam}); err != nil {
+				return reconcile.Result{}, err
+			}
```

Comment:
Did you verify that with our existing 200-role.yaml? I'd assumed we need to add update permission for taskruns?

Reply:
Yeah I think I lost that in my rebase somehow.  I was doing some manual testing on openshift just know and ran into this.

Per the results I had in the description, I had them at some point, otherwise, I would not have been able to get those results.

Will update shortly - thanks.

Reply:
updated